Tudat/SimulationSetup/body.h

```diff
@@ -0,0 +1,300 @@
+/*    Copyright (c) 2010-2016, Delft University of Technology
+ *    All rigths reserved
+ *
+ *    This file is part of the Tudat. Redistribution and use in source and
+ *    binary forms, with or without modification, are permitted exclusively
+ *    under the terms of the Modified BSD license. You should have received
+ *    a copy of the license with this file. If not, please or visit:
+ *    http://tudat.tudelft.nl/LICENSE.
+ */
+
+#ifndef TUDAT_BODY_H
+#define TUDAT_BODY_H
+
+#include <map>
+#include <vector>
+
+#include <boost/shared_ptr.hpp>
+
+#include <Eigen/Core>
+
+#include <Tudat/Astrodynamics/Aerodynamics/atmosphereModel.h>
+#include <Tudat/Astrodynamics/BasicAstrodynamics/timeConversions.h>
+#include <Tudat/Astrodynamics/BasicAstrodynamics/accelerationModel.h>
+#include <Tudat/Astrodynamics/Ephemerides/ephemeris.h>
+#include <Tudat/Astrodynamics/Ephemerides/rotationalEphemeris.h>
+#include <Tudat/Astrodynamics/Gravitation/gravityFieldModel.h>
+#include <Tudat/Mathematics/BasicMathematics/linearAlgebraTypes.h>
+#include <Tudat/Astrodynamics/Ephemerides/rotationalEphemeris.h>
+
+namespace tudat
+{
+
+namespace simulation_setup
+{
+
+//! Body class representing the properties of a celestial body (natural or artificial).
+/*!
+ *  Body class representing the properties of a celestial body (natural or artificial). By storing
+ *  all properties of bodies (ephemeris, rotation, gravity, etc.) in a set of body objects,
+ *  the simulation environment can be defined in a clear and modular way. To create body
+ *  objects, the createBodies.h function provides a range of functionality. The
+ *  createAccelerationModels.h file provides functions to use body objects to create acceleration
+ *  objects.
+ */
+class Body
+{
+public:
+
+    //! Constructor for a body
+    /*!
+     * Constructor for a body, sets current time, state, rotation and mass values
+     * (all with default parameters). The input state is used internally to
+     * set the current position (taken as a segment of the input state given by the indices
+     * (0, 3)) and the current velocity (taken as a segment of the input state given by the indices
+     * (3, 3).
+     * \param state Current state of body at initialization (default = zeroes).
+     * \param time Current time of body at initialization (default = zeroes).
+     * \param bodyMass Current mass of body at initialization (default = zeroes).
+     * \param currentRotationToGlobalFrame Current rotation of from body-fixed to inertial frames
+     *  at initialization (default = identity)
+     */
+    Body( const basic_mathematics::Vector6d& state =
+            basic_mathematics::Vector6d::Zero( ),
+          const double time = 0.0, const double bodyMass = 0.0,
```

Comment:
Why is the "bodyMass" even part of the constructor? There is no return or other functions using it, and any value is overridden by setGravityFieldModel.